### PR TITLE
Add getting started & integration guides

### PR DESCRIPTION
**Pull request by cardoso**

### CODE REVIEW

File: stream-chat-swift-docs/docs/guides/integration.md [hunk 1]
```diff
@@ -2,8 +2,43 @@
...
+
+```ruby
+target 'ChatDemo' do
+  # Comment the next line if you don't want to use dynamic frameworks
+  use_frameworks!
+
+  # Pods for ChatDemo
+  pod 'StreamChatUI', '~> 3.1'
+end
+```
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/integration.md:19]: Can you add a section for just installing `StreamChat`?

File: stream-chat-swift-docs/docs/guides/integration.md [hunk 2]
```diff
@@ -2,8 +2,43 @@
...
+pod install --repo-update
+```
+
+The above command will generate the **ChatDemo.xcworkspace** file automatically.
+
+With our workspace now containing our Pods project with dependencies, as well as our original project, let’s go ahead and move over to Xcode to complete the process.
+
+## Swift Package Manager
+
+Add the following to your Package.swift or in Xcode -> File -> Swift Packages -> Add Package Dependency:
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/integration.md:33]: Can you put a screenshot?

File: stream-chat-swift-docs/docs/guides/integration.md [hunk 3]
```diff
@@ -2,8 +2,43 @@
...
+
+```swift
+dependencies: [
+    .package(url: "https://github.com/GetStream/stream-chat-swift.git", .upToNextMajor(from: "3.1"))
+]
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/integration.md:38]: If Xcode UI is used, the code snippet is unnecessary, it'd be nice if you show the Xcode UI

File: stream-chat-swift-docs/docs/guides/getting-started.md [hunk 4]
```diff
@@ -1,3 +1,149 @@
...
+    if let token = token {
+        completion(.success(token))
+    } else if let error = error {
+        completion(.failure(error))
+    }
+}
+```
+
+#### Development
+
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/getting-started.md:45]: "To use this, it must be enabled" as you've linked above, the info should be here

File: stream-chat-swift-docs/docs/guides/getting-started.md [hunk 5]
```diff
@@ -1,3 +1,149 @@
...
+## User Authentication
+
+After the SDK is installed, the next step is to initialize the chat client and authenticate the user. It can be a regular user, or it can be an [anonymous](https://getstream.io/chat/docs/anon/?language=swift) or [guest](https://getstream.io/chat/docs/guest_users/?language=swift) user with limited sets of permissions. A `TokenProvider` is the object that will hold your authentication strategy and will tell the `ChatClient` which user you're currently logged in with.
+
+## User Tokens
+
+User Tokens are JWT tokens containing a User ID and used to authenticate a user. You can [generate these tokens using a server client](https://getstream.io/chat/docs/ios-swift/tokens_and_authentication/?language=swift#generating-tokens) and then use them to authenticate a user in a chat client.
+
+### Regular User
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/getting-started.md:13]: I don't find "Regular" a good adjective here but I can't come up with a better one...

File: stream-chat-swift-docs/docs/guides/getting-started.md [hunk 6]
```diff
@@ -1,3 +1,149 @@
...
+### Anonymous
+
+Anonymous users don't need a special token or any identification, but they can't do much except reading livestream chats. They're ideal for the livestream use case before the user wants to identify themselves and interact.
+
+```swift
+/// An anonymous token provider. Use it to let the user see livestream chats without identifying themselves or creating an account.
+let tokenProvider = TokenProvider.anonymous
+```
+
+## Configuration
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/getting-started.md:69]:
```suggestion
## ChatClient Configuration
```

File: stream-chat-swift-docs/docs/guides/getting-started.md [hunk 7]
```diff
@@ -1,3 +1,149 @@
...
+
+To make a `ChatClient` singleton, we extend the `ChatClient` class with a static shared instance of itself.
+
+```swift
+/// 1: Extend the `ChatClient` class.
+extension ChatClient {
+
+    /// 2: Add a `shared` static variable of its own type.
+    static var shared: ChatClient!
+}
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/getting-started.md:87]: This snippet doesn't work if the developer created their own ExtraData and defined typealises. Even when the typealises are defined, the extension extends the default (`_ChatClient<NoExtraData>`) type
```swift
extension _ChatClient {
```
works better, but needs testing if it works for the default case too

File: stream-chat-swift-docs/docs/guides/getting-started.md [hunk 8]
```diff
@@ -1,3 +1,149 @@
...
+let tokenProvider = TokenProvider.anonymous
+```
+
+## Configuration
+
+The next step is to configure a `ChatClient` instance with your API Key and the `tokenProvider` from the previous step. The most simple way to do this is by extending the `ChatClient` class with a shared instance that will be used throughout your app (Singleton). It's also possible to create a chat instance in a convenient entry point and pass it down to your classes / view controllers. (Dependency Injection).
+
+### ChatClientConfig
+
+The `ChatClientConfig` object holds properties that the chat client will use to determine certain behaviors. For example, the `apiKey`, which tells the chat client which chat server to communicate with. It also has the `baseURL` property which tells the chat client which region of the world your server is at, which can be useful to reduce overall latency.
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/getting-started.md:75]: Needs a code snippet to show how `baseURL` is changed, preferably also showing the dashboard so users can configure the correct baseURL

File: stream-chat-swift-docs/docs/guides/integration.md [hunk 9]
```diff
@@ -2,8 +2,61 @@
...
+
+After pressing next, Xcode will look for the repository and automatically select the latest version tagged. Press next and Xcode will download the dependency.
+
+![Screenshot shows an Xcode screen selecting a dependency version and an Xcode screen downloading that dependency](/img/spm-01.png)
+
+The repository contains 2 targets: StreamChat and StreamChatUI. If you'll use the UI components, select both. If you don't need the UI components, select just StreamChat.
+
+![Screenshot shows an Xcode screen with dependency targets to be selected](/img/spm-02.png)
+
+After you press finish, it's done!
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/integration.md:57]: kudos ✅

File: stream-chat-swift-docs/docs/guides/getting-started.md [hunk 10]
```diff
@@ -1,3 +1,149 @@
...
+class SceneDelegate: UIResponder, UIWindowSceneDelegate {
+    var window: UIWindow?
+
+    func scene(
+        _ scene: UIScene,
+        willConnectTo session: UISceneSession,
+        options connectionOptions: UIScene.ConnectionOptions
+    ) {
+        guard let scene = (scene as? UIWindowScene) else { return }
+        scene.windows.forEach { $0.tintColor = .systemPink }
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/getting-started.md:134]: Let's not put this line here

File: stream-chat-swift-docs/docs/guides/getting-started.md [hunk 11]
```diff
@@ -1,3 +1,149 @@
...
+
+        /// 1: Get the reference to the initial view controller
+        let viewController = window?.rootViewController as? ViewController
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/getting-started.md:137]: We should put the simplest ViewController definition up top, so copy/paste/compile works

**Review by b-onc — CHANGES_REQUESTED** [review]

File: stream-chat-swift-docs/docs/guides/getting-started.md (hunk 7 above)
Comment (cardoso) [reply stream-chat-swift-docs/docs/guides/getting-started.md:87]: Didn't work: static stored properties not supported in generic types.

File: stream-chat-swift-docs/docs/guides/getting-started.md [hunk 12]
```diff
@@ -1,3 +1,165 @@
...
+
+The next step is to configure a `ChatClient` instance with your API Key and the `tokenProvider` from the previous step. The most simple way to do this is by extending the `ChatClient` class with a shared instance that will be used throughout your app (Singleton). It's also possible to create a chat instance in a convenient entry point and pass it down to your classes / view controllers. (Dependency Injection).
+
+### ChatClientConfig
+
+The `ChatClientConfig` object holds properties that the chat client will use to determine certain behaviors. For example, the `apiKey`, which tells the chat client which chat server to communicate with. It also has the `baseURL` property which tells the chat client which region of the world your server is at, which can be useful to reduce overall latency.
+
+```swift
+/// 1: Create a `ChatClientConfig` instance with the API key.
+let config = ChatClientConfig(apiKeyString: "YOUR_API_KEY")
```
Comment (b-onc) [inline stream-chat-swift-docs/docs/guides/getting-started.md:81]:
```suggestion
var config = ChatClientConfig(apiKeyString: "YOUR_API_KEY")
```

Comment (b-onc) [reply stream-chat-swift-docs/docs/guides/getting-started.md:81]: else compiler will complain

**Comment by b-onc** [comment]
Continuing in #1142

**Pull request closed without merging**